### PR TITLE
Flatlists and UI tweaks

### LINKED ISSUE #343: Convert ScrollViews to FlastLists
The scrollable content in the threads list and thread detail list are both plain scroll views. This results in all the displayed data/photos being queries and rendered even if they aren't on screen.

### PR DESCRIPTION
**Pull request by andrewxhill**
Fixes #343

### CODE REVIEW

**Comment by andrewxhill** [comment]
@asutula I wanted to avoid conflicts so am pointing this at your branch. merge whenever.

File: App/Containers/ThreadsList.js [hunk 1]
```diff
@@ -109,18 +117,13 @@ class ThreadsList extends React.PureComponent {
         )}
         {this.props.threads.length !== 0 && (
           // FIXME: This should be a FlatList for sure
-          <ScrollView
-            style={styles.contentContainer}
-            refreshControl={
-              <RefreshControl
```
Comment (asutula) [inline App/Containers/ThreadsList.js]: Not sure if you wanted to drop refresh here, but FlatList supports it if you want it.

Comment (andrewxhill) [reply App/Containers/ThreadsList.js]: ah for sure, will add back

File: App/SB/components/ThreadListCard/index.js [hunk 2]
```diff
@@ -9,14 +9,22 @@ import Avatar from '../../../Components/Avatar'
...
-  if (props.profile && props.profile.id && props.profile.id === latestPeerId) {
-    uri = 'https://cafe.us-east-1.textile.io' + props.profile.avatar_id
+  let title = name
+  if (title.length > 17) {
+    title = title.substring(0, 32)
+    title = title.substring(0, title.lastIndexOf(' '))
+    while (title[title.length - 1] === '.' || title[title.length - 1] === ' ') {
+      title = title.slice(0, -1)
+    }
+    title += '...'
```
Comment (asutula) [inline App/SB/components/ThreadListCard/index.js]: Surprised if there is no support for automatically formatting text like this when the test is longer than the view is wide. It exists in iOS for sure, probably Android too.

Comment (asutula) [reply App/SB/components/ThreadListCard/index.js]: https://facebook.github.io/react-native/docs/text#ellipsizemode

Comment (andrewxhill) [reply App/SB/components/ThreadListCard/index.js]: cool, already added

**Pull request merged**